### ISSUE TITLE
valid-expect - Triggers error in Custom Matcher code example from Playwright's doc

### ISSUE THREAD

**Issue by mchlroy** [mskelton/eslint-plugin-playwright#405]
It's not a huge issue, but I thought I'd point it out anyway, there could be a way to more accurately check if the `expect` eventually calls a `matcher`, but up to you guys to see if it's worth it.

<img width="1355" height="263" alt="Image" src="https://github.com/user-attachments/assets/d0e6b466-cc49-46e0-bece-9a33724fd72a" />

Of course, one can just change the code to:
```ts
if (this.isNot) {
    await baseExpect(locator).not.toHaveAttribute('data-amount', String(expected), options);
} else {
    await baseExpect(locator).toHaveAttribute('data-amount', String(expected), options);
}
```
to get rid of the linter error. 

Considering this is straight from the [doc](https://playwright.dev/docs/test-assertions#add-custom-matchers-using-expectextend), I thought I'd let you guys know. Feel free to close this as _won't fix_ . I personally prefer the _if / else_ rather than _ternary_ here anyways.

**Comment by mskelton** [comment]
Since the example is inside a custom matcher, I'm fine with this being eslint-disable-next-line
